<a href="https://colab.research.google.com/github/lindseyvanosky/machine_learning/blob/main/LV_Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##Data Set #1 - Leafly Strain Data

###Import

In [3]:
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')
#import
leafly_data = "/content/drive/MyDrive/Coding Dojo/02 Stack 2: Intro to Machine Learning/Data Sets/leafly_strain_data.csv"
leafly_df = pd.read_csv(leafly_data)
leafly_df.head()

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


name                                            img_url    type  \
0           GG4   https://images.leafly.com/flower-images/gg-4.jpg  Hybrid   
1  Wedding Cake  https://images.leafly.com/flower-images/weddin...  Hybrid   
2         Runtz  https://images.leafly.com/flower-images/runtz-...  Hybrid   
3        Gelato  https://images.leafly.com/flower-images/gelato...  Hybrid   
4    Blue Dream  https://images.leafly.com/flower-images/blue-d...  Hybrid   

  thc_level most_common_terpene  \
0       20%       Caryophyllene   
1       22%            Limonene   
2       19%            Limonene   
3       17%       Caryophyllene   
4       18%             Myrcene   

                                         description relaxed happy euphoric  \
0  Original Glue (GG4), developed by GG Strains, ...     66%   54%      51%   
1  Wedding Cake is a potent indica-hybrid marijua...     60%   50%      41%   
2  Runtz is a rare hybrid marijuana strain by Coo...     47%   62%      48%   
3  Gelato, aka Larry Bird, is a hybrid marijuana ...     54%   49%      46%   
4  Blue Dream is a sativa-dominant hybrid marijua...     48%   57%      52%   

  uplifted  ... fibromyalgia crohn's_disease phantom_limb_pain epilepsy  \
0      39%  ...           0%              0%                0%       0%   
1      35%  ...           0%              0%                0%       0%   
2      43%  ...           0%              0%                0%       0%   
3      37%  ...           0%              0%                0%       0%   
4      48%  ...           0%              0%                0%       0%   

  multiple_sclerosis parkinson's tourette's_syndrome alzheimer's hiv/aids  \
0                 0%          0%                  0%          0%       0%   
1                 0%          0%                  0%          0%       0%   
2                 0%          0%                  0%          0%       0%   
3                 0%          0%                  0%          0%       0%   
4                 0%          0%                  0%          0%       0%   

  tinnitus  
0       0%  
1       0%  
2       0%  
3       0%  
4       0%  

[5 rows x 64 columns]

###Info

In [4]:
leafly_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4762 entries, 0 to 4761
Data columns (total 64 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   name                       4762 non-null   object
 1   img_url                    98 non-null     object
 2   type                       4107 non-null   object
 3   thc_level                  2735 non-null   object
 4   most_common_terpene        2447 non-null   object
 5   description                4727 non-null   object
 6   relaxed                    4762 non-null   object
 7   happy                      4762 non-null   object
 8   euphoric                   4762 non-null   object
 9   uplifted                   4762 non-null   object
 10  sleepy                     4762 non-null   object
 11  dry_mouth                  4762 non-null   object
 12  dry_eyes                   4762 non-null   object
 13  dizzy                      4762 non-null   object
 14  paranoid

###Questions

Source: [Kaggle.](https://www.kaggle.com/datasets/gthrosa/leafly-cannabis-strains-metadata) This data set is public domain.

Description: This data set is a complete list of cannabis strains and various attributes. 

Target: I'd like to investigate two targets: THC % and type.

Classification or Regression? I'd like to run both as I would use a regression model to predict THC % and I would use a Classisfication model to determine a strain type (sativa, hybrid, indica)

Features: 4, most common terpene, type, thc level, description(effects)

How many rows? 4762

Potential Challenges: One potential challenge with this data set is that there is a lot of overlap of terpenes and effects between strains. Additionally all of the dtypes are objects, so OHE could make this data set too large. 

##Data Set #2 - American Demographics

###Import

In [8]:
#import
census_data = "/content/drive/MyDrive/Coding Dojo/02 Stack 2: Intro to Machine Learning/Data Sets/adult.data"
census_df = pd.read_csv(census_data, header = None)
census_df.head()

0                  1       2           3   4                    5   \
0  39          State-gov   77516   Bachelors  13        Never-married   
1  50   Self-emp-not-inc   83311   Bachelors  13   Married-civ-spouse   
2  38            Private  215646     HS-grad   9             Divorced   
3  53            Private  234721        11th   7   Married-civ-spouse   
4  28            Private  338409   Bachelors  13   Married-civ-spouse   

                   6               7       8        9     10  11  12  \
0        Adm-clerical   Not-in-family   White     Male  2174   0  40   
1     Exec-managerial         Husband   White     Male     0   0  13   
2   Handlers-cleaners   Not-in-family   White     Male     0   0  40   
3   Handlers-cleaners         Husband   Black     Male     0   0  40   
4      Prof-specialty            Wife   Black   Female     0   0  40   

               13      14  
0   United-States   <=50K  
1   United-States   <=50K  
2   United-States   <=50K  
3   United-States   <=50K  
4            Cuba   <=50K

###Info

In [7]:
census_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32560 entries, 0 to 32559
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   39              32560 non-null  int64 
 1    State-gov      32560 non-null  object
 2    77516          32560 non-null  int64 
 3    Bachelors      32560 non-null  object
 4    13             32560 non-null  int64 
 5    Never-married  32560 non-null  object
 6    Adm-clerical   32560 non-null  object
 7    Not-in-family  32560 non-null  object
 8    White          32560 non-null  object
 9    Male           32560 non-null  object
 10   2174           32560 non-null  int64 
 11   0              32560 non-null  int64 
 12   40             32560 non-null  int64 
 13   United-States  32560 non-null  object
 14   <=50K          32560 non-null  object
dtypes: int64(6), object(9)
memory usage: 3.7+ MB


###Questions

Source: [Kaggle](https://www.kaggle.com/datasets/danielbethell/adult-incomes-in-the-united-states?select=adult.names)
<br>
This data was extracted from the census bureau [database.](http://www.census.gov/ftp/pub/DES/www/welcome.html)

Description: This data set contains demographics on the American people in order to determine if they make more that $50K



Target: Income

Classification or Regression? Regression! 

Features: Age, Sex, Education, Job Type, Family, Race

How many rows? 32560

Potential Challenges: One challenge I see with this data set is that it does not have a header, so we will have to rename each column to make it make sense. 